### PR TITLE
Insert mode

### PR DESCRIPTION
**Pull request by albfan**
## Summary / How this PR fixes the problem?
This PR start basic workflow on Akira. Still needs lot of refinement and deal with corner cases to be intuitive but basically:

<kbd>C</kbd>,<kbd>R</kbd>,<kbd>T</kbd> go to insert mode and add Circle, Rectangle and text respectively
<kbd>Esc</kbd> return to selection mode
<kbd>Supr</kbd> delete selected object

## Steps to Test
Play around with shorcuts or menus

## Screenshots 
insert-mode

## Known Issues / Things To Do

Interact with text is still weird, but I think is a known problem.
In insert mode you cannot interact with existing objects. I think sometimes user expects to exit insert mode clicking on existing objects (or double clicking)

## This PR fixes/implements the following **bugs/features**:
There're no open issues for this

### CODE REVIEW

**Comment by Alecaddd** [comment]
This is a great start and I love it so much!
A couple of things to fix before merging:

- After an object has been drew, the canvas should revert back to `EDIT_MODE`. Let's not keep the insert tool active.
- These Canvas shortcut, should be triggered only if the Canvas has focus. If the focus is on an input field, like while renaming a layer or an artboard, it's currently impossible to type those letters.
- The canvas should grab the focus when the app is launched.

With this few changes this is good to merge IMO.
Amazing work!

**Comment by Fatih20** [comment]
What if instead of just keyboard shortcut and dropdown menu to add shapes, we can right click on the canvas that then spawn a window that has the options to add shapes? I would refer to https://github.com/phase1geo/minder/ if you want to know how this would work.

**Comment by albfan** [comment]
@Fatih20 context menu makes sense to me. @Alecaddd just let me know

**Comment by Fatih20** [comment]
> @Fatih20 context menu makes sense to me. @Alecaddd just let me know

Well, you don't have to get rid of context menu , you can add this functionality if you decide that this is good and still have context menu.

**Comment by albfan** [comment]
@Fatih20 "right click on the canvas that then spawn a window" is the definition of a context menu. We don't have such thing (so there's nothing to get rid of) instead your suggestion is to add it, isn't it?

NOTE: Point to a hole project for a specific functionality is hard to follow. Take some screenshots or screencast. I suggest to do on a new issue

**Comment by Fatih20** [comment]
image
Here's an example.

**Comment by Alecaddd** [comment]
That's a good suggestion, but not for this PR.
We will slowly add context menus for specific sections, but for now we're focusing on implementing features accessible via the main UI.

**Comment by albfan** [comment]
- [x] After an object has been drew, the canvas should revert back to EDIT_MODE. Let's not keep the insert tool active.

> Yes, feels more natural this way

- [x] These Canvas shortcut, should be triggered only if the Canvas has focus. If the focus is on an input field, like while renaming a layer or an artboard, it's currently impossible to type those letters.

> Didn't realize that, fixed

- [x] The canvas should grab the focus when the app is launched.

> Now you can start to draw once app is started without click on canvas to grab the focus

File: src/Lib/Canvas.vala [hunk 1]
```diff
@@ -113,18 +160,33 @@ public class Akira.Lib.Canvas : Goo.Canvas {
...
         } else {
             remove_select_effect ();
-            grab_focus (get_root_item ());
+            focus ();
         }
 
         return true;
     }
 
+    public void focus () {
```
Comment (Alecaddd) [inline src/Lib/Canvas.vala]: We should add `new` after `public` since we're overriding an inherited method.

File: src/Services/ActionManager.vala [hunk 2]
```diff
@@ -154,25 +158,28 @@ public class Akira.Services.ActionManager : Object {
...
+        window.main_window.main_canvas.canvas.insert_type = null;
     }
 
-    private void action_add_text () {
-        var text = window.main_window.main_canvas.add_text ();
-        var artboard = window.main_window.right_sidebar.layers_panel.artboard;
-        artboard.container.add (new Akira.Layouts.Partials.Layer (window, artboard, text, "Text", "shape-text-symbolic", false));
-        artboard.show_all ();
+    private void action_delete () {
+        window.main_window.main_canvas.canvas.delete_selected ();
```
Comment (Alecaddd) [inline src/Services/ActionManager.vala]: `action_delete` doesn't seem to work anymore.

File: src/Layouts/Partials/Layer.vala [hunk 3]
```diff
@@ -518,7 +518,7 @@ public class Akira.Layouts.Partials.Layer : Gtk.ListBoxRow {
     }
 
     public bool update_on_escape (Gdk.EventKey key) {
-        if (key.keyval == 65307) {
+        if (key.keyval == Gdk.Key.Search) {
```
Comment (Alecaddd) [inline src/Layouts/Partials/Layer.vala]: This should be the `Esc` key.
If you're changing this, please update also the same method in the `Artboard.vala` file

**Comment by Alecaddd** [comment]
I took care of the small issues. I'll merge this soon
